src/probnum/statespace/sde.py

```diff
@@ -248,6 +298,39 @@ def f(t, y):
 
         return f, y0
 
+    def _setup_vectorized_mde_backward(self, initrv_obtained, _diffusion=1.0):
+        """Set up backward moment differential equations (MDEs).
+
+        Compute an ODE vector field that represents the MDEs and is
+        compatible with scipy.solve_ivp.
+        """
+        dim = len(initrv_obtained.mean)
+
+        def f(t, y, mde_forward_sol_mean, mde_forward_sol_cov):
+            # Undo vectorization
+            mean, cov_flat = y[:dim], y[dim:]
+            cov = cov_flat.reshape((dim, dim))
+
+            # Apply iteration
+            F = self.driftmatfun(t)
+            u = self.forcevecfun(t)
+            L = self.dispmatfun(t)
+
+            LL = _diffusion * L @ L.T
+            LL_inv_cov = LL @ np.linalg.inv(mde_forward_sol_cov(t))
+
+            new_mean = F @ mean + LL_inv_cov @ (mean - mde_forward_sol_mean(t)) + u
```

Comment:
Looking more closely at the differential equations I don't see how one could use `solve`, since e.g. the smoothing-covariance matrix is some kind of Lyapunov equation. Did I miss something?

Reply:
- Good idea.
- I use `solve` as you suggested